app/assets/javascripts/discourse/app/components/modal/avatar-selector.js

```diff
@@ -0,0 +1,185 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { allowsImages } from "discourse/lib/uploads";
+import { isTesting } from "discourse-common/config/environment";
+
+export default class AvatarSelectorModal extends Component {
+  @service currentUser;
+  @service siteSettings;
+  @tracked gravatarRefreshDisabled = false;
+  @tracked gravatarFailed = false;
+  @tracked uploading = false;
+  @tracked _selected = null;
+
+  get gravatar() {
```

Comment:
If this is used only to provide values to i18n strings, I'd inline those and remove this getter.